### ISSUE TITLE
Remove setup_bazel.sh in favor of bazel autoconfiguration [phab:D760]

### ISSUE THREAD

**Issue by kythe** [google/kythe#1043]
_Created by **shahms** at 2016-03-07 22:56:28_: 

Use bazel configuration for node, libmemcached, uuid and libcrypto.

This setup also remove the flags
            "-Wl,-Bstatic",
            "-lmemcached",
            "-Wl,-Bdynamic",
            "-lsasl2",
from libmemcached and I don't know why/if they were previously necessary.

Note: Please test this under OS X!

**Comment by kythe** [comment]
_**shahms** wrote at 2016-03-07 23:14:50_:

  - Fix for  attr paths

**Comment by kythe** [comment]
_**shahms** wrote at 2016-03-08 07:15:49_:

  - Properly reference headers and include paths.

**Comment by kythe** [comment]
_**schroederc** accepted this revision at 2016-03-10 20:28:47_

**Comment by kythe** [comment]
_**schroederc** wrote at 2016-03-10 20:29:40_:

http://www.kythe.io/contributing and http://www.kythe.io/2015/06/05/osx-support/ need fixing

**Comment by kythe** [comment]
_**zarko** wrote at 2016-03-10 20:31:12_:

OS X fails:
```
WARNING: /Users/zarko/kythe-ci/tools/node_configure.bzl:10:5: No node.js installation found.
ERROR: /private/var/tmp/_bazel_zarko/2be6994bf77f75bac8f1433eb6e9dc33/external/local_libmemcached/BUILD:7:14: in linkopts attribute of cc_library rule local_libmemcached//:libmemcached: could not resolve label 'external/local_libmemcached/lib'.
ERROR: /private/var/tmp/_bazel_zarko/2be6994bf77f75bac8f1433eb6e9dc33/external/local_libmemcached/BUILD:7:14: in linkopts attribute of cc_library rule local_libmemcached//:libmemcached: could not resolve label 'external/local_libmemcached/lib'.
ERROR: Analysis of target '//kythe/cxx/common/indexing:testlib' failed; build aborted.
INFO: Elapsed time: 6.989s
```

**Comment by kythe** [comment]
_**zarko** required changes at 2016-03-10 20:31:12_

**Comment by kythe** [comment]
_**shahms** wrote at 2016-03-10 20:36:19_:

>>! In (earlier comment), zarko wrote:
> OS X fails:
> ```
> WARNING: /Users/zarko/kythe-ci/tools/node_configure.bzl:10:5: No node.js installation found.
> ERROR: /private/var/tmp/_bazel_zarko/2be6994bf77f75bac8f1433eb6e9dc33/external/local_libmemcached/BUILD:7:14: in linkopts attribute of cc_library rule local_libmemcached//:libmemcached: could not resolve label 'external/local_libmemcached/lib'.
> ERROR: /private/var/tmp/_bazel_zarko/2be6994bf77f75bac8f1433eb6e9dc33/external/local_libmemcached/BUILD:7:14: in linkopts attribute of cc_library rule local_libmemcached//:libmemcached: could not resolve label 'external/local_libmemcached/lib'.
> ERROR: Analysis of target '//kythe/cxx/common/indexing:testlib' failed; build aborted.
> INFO: Elapsed time: 6.989s
> ```

Do you have NODEJS and MEMCACHED_HOME environment variables defined?

**Comment by kythe** [comment]
_**shahms** wrote at 2016-03-10 20:37:37_:

Doh! Nevermind, I've got a fix for the memcached issue.

**Comment by kythe** [comment]
_**shahms** wrote at 2016-03-10 20:38:36_:

  - Prepend -L to the lib directory in linkopts on OS X

**Comment by kythe** [comment]
_**zarko** wrote at 2016-03-10 20:43:50_:

Fixed it. Thanks!

**Comment by kythe** [comment]
_**zarko** accepted this revision at 2016-03-10 20:43:50_

**Comment by kythe** [comment]
_**shahms** wrote at 2016-03-10 20:50:55_:

  - Update documentation to remove setup_bazel.sh

**Comment by kythe** [comment]
_**shahms** closed via commit f3c472a0e6b75799dd2bcddf8b18c7ac56af381f at 2016-03-10 21:20:08_
